File: 0835-image-overlap/0835-image-overlap.py
class Solution:
    def largestOverlap(self, img1: list[list[int]], img2: list[list[int]]) -> int:
        n = len(img1)
        size = 2 * n - 1

        # Correct initialization for 2D list in Python
        count = [[0] * size for _ in range(size)]
        answer = 0

        for r1 in range(n):
            for c1 in range(n):
                if img1[r1][c1] == 0:
                    continue

                for r2 in range(n):
                    for c2 in range(n):
                        if img2[r2][c2] == 0:
                            continue

                        dr = r2 - r1 + n - 1
                        dc = c2 - c1 + n - 1

                        count[dr][dc] += 1
                        if count[dr][dc] > answer:
                            answer = count[dr][dc]

        return answer

File: 0835-image-overlap/0835-image-overlap.js
/**
 * @param {number[][]} img1
 * @param {number[][]} img2
 * @return {number}
 */
var largestOverlap = function(img1, img2) {
    const n = img1.length;
    const size = 2 * n - 1;

    // Create a 2D array of size x size filled with zeros
    const count = Array.from({ length: size }, () => new Array(size).fill(0));
    let answer = 0;

    for (let r1 = 0; r1 < n; r1++) {
        for (let c1 = 0; c1 < n; c1++) {
            if (img1[r1][c1] === 0) continue;

            for (let r2 = 0; r2 < n; r2++) {
                for (let c2 = 0; c2 < n; c2++) {
                    if (img2[r2][c2] === 0) continue;

                    const dr = r2 - r1 + n - 1;
                    const dc = c2 - c1 + n - 1;

                    count[dr][dc]++;
                    answer = Math.max(answer, count[dr][dc]);
                }
            }
        }
    }

    return answer;
};
